pkg/controller/jobframework/tas_test.go

```diff
@@ -0,0 +1,228 @@
+					kueuealpha.PodSetSliceSizeAnnotation:             "1",
+				},
+			},
+			wantReq: &kueue.PodSetTopologyRequest{
+				PodSetSliceRequiredTopology: ptr.To("cloud.com/block"),
+				PodSetSliceSize:             ptr.To[int32](1),
+			},
+		},
+		"slice-only topology – only slice required annotation": {
+			meta: &metav1.ObjectMeta{
+				Annotations: map[string]string{
+					kueuealpha.PodSetSliceRequiredTopologyAnnotation: "cloud.com/block",
+				},
+			},
+		},
+		"slice-only topology – only slice size annotation": {
+			meta: &metav1.ObjectMeta{
+				Annotations: map[string]string{
+					kueuealpha.PodSetSliceSizeAnnotation: "1",
+				},
+			},
+		},
+		"slice-only topology with sub group and pod set group name annotation": {
+			meta: &metav1.ObjectMeta{
+				Annotations: map[string]string{
+					kueuealpha.PodSetSliceRequiredTopologyAnnotation: "cloud.com/block",
+					kueuealpha.PodSetSliceSizeAnnotation:             "1",
+					kueuealpha.PodSetGroupName:                       "block",
+				},
+			},
+			subGroupIndexLabel: ptr.To(jobsetapi.JobIndexKey),
+			subGroupCount:      ptr.To[int32](1),
+			wantReq: &kueue.PodSetTopologyRequest{
+				PodSetSliceRequiredTopology: ptr.To("cloud.com/block"),
+				PodSetSliceSize:             ptr.To[int32](1),
+				SubGroupIndexLabel:          ptr.To(jobsetapi.JobIndexKey),
+				SubGroupCount:               ptr.To[int32](1),
+			},
+		},
+		"invalid unconstrained topology annotation value": {
+			meta: &metav1.ObjectMeta{
+				Annotations: map[string]string{
+					kueuealpha.PodSetUnconstrainedTopologyAnnotation: "invalid",
+				},
+			},
+			wantErr: &strconv.NumError{Func: "ParseBool", Num: "invalid", Err: strconv.ErrSyntax},
+		},
+		"invalid podset slice size annotation value": {
+			meta: &metav1.ObjectMeta{
+				Annotations: map[string]string{
+					kueuealpha.PodSetSliceRequiredTopologyAnnotation: "cloud.com/block",
+					kueuealpha.PodSetSliceSizeAnnotation:             "invalid",
+				},
+			},
+			wantErr: &strconv.NumError{Func: "ParseInt", Num: "invalid", Err: strconv.ErrSyntax},
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			b := NewPodSetTopologyRequest(tc.meta)
+			b.PodIndexLabel(tc.podIndexLabel)
+			b.SubGroup(tc.subGroupIndexLabel, tc.subGroupCount)
+
+			gotReq, gotErr := b.Build()
+
+			if diff := cmp.Diff(tc.wantReq, gotReq); len(diff) != 0 {
+				t.Errorf("Unexpected request (-want,+got):\n%s", diff)
+			}
+
+			if diff := cmp.Diff(tc.wantErr, gotErr,
+				// Filter to apply EquateErrors only to non-*strconv.NumError types
+				cmp.FilterPath(func(path cmp.Path) bool {
+					return path.Last().Type() != reflect.TypeOf(&strconv.NumError{})
+				}, cmpopts.EquateErrors()),
+				// Filter to apply numErrorComparer only to *strconv.NumError
+				cmp.FilterPath(func(path cmp.Path) bool {
+					return path.Last().Type() == reflect.TypeOf(&strconv.NumError{})
+				}, numErrorComparer),
```

Comment:
Why can't we rely on cmpopts.EquateErrors as above and we implement a custom comparator?

Reply:
Done